rskj-core/src/main/java/co/rsk/trie/NodeReference.java

```diff
@@ -55,7 +54,11 @@ public boolean isEmpty() {
      */
     public void save() {
         if (lazyNode != null) {
-            lazyNode.save();
+            if (!isEmbeddable()) {
+                lazyNode.save();
+            } else {
+                store.saveValue(lazyNode);
```

Comment:
This is called even when the node doesn't have a long value. I think this is a bug.